### ISSUE TITLE
error installing seurat 3 alpha

### ISSUE THREAD

**Issue by orrzor** [satijalab/seurat#914]
Hi,

Thanks for releasing the new Seurat version for testing! I am trying to install it and getting an issue with devtools. I don't know if this is an issue I should ask you or the devtools team. Thanks a lot! -Orr

```r
> devtools::install_github(repo = 'satijalab/seurat', ref = 'release/3.0')
Downloading GitHub repo satijalab/seurat@release/3.0
Installing 7 packages: future, future.apply, ggrepel, globals, listenv, reticulate, rsvd
Error: (converted from warning) unable to access index for repository https://mojaveazure.github.io/loomR/bin/macosx/el-capitan/contrib/3.5:
  cannot open URL 'https://mojaveazure.github.io/loomR/bin/macosx/el-capitan/contrib/3.5/PACKAGES'
> sessionInfo()
R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.19      rstudioapi_0.8    magrittr_1.5      usethis_1.4.0    
 [5] devtools_2.0.1    pkgload_1.0.2     R6_2.3.0          rlang_0.3.0.1    
 [9] tools_3.5.1       pkgbuild_1.0.2    sessioninfo_1.1.0 cli_1.0.1        
[13] withr_2.1.2       remotes_2.0.2     assertthat_0.2.0  digest_0.6.18    
[17] rprojroot_1.3-2   crayon_1.3.4      processx_3.2.0    callr_3.0.0      
[21] base64enc_0.1-3   fs_1.2.6          ps_1.2.0          curl_3.2         
[25] memoise_1.1.0     glue_1.3.0        compiler_3.5.1    desc_1.2.0       
[29] backports_1.1.2   prettyunits_1.0.2
```

Also for what it's worth, I have `loomR` already.
```r
> library(loomR)
Loading required package: R6
Loading required package: hdf5r
Loading required package: iterators
Loading required package: itertools
```
And this command for regular Seurat works
```r
devtools::install_version(package = 'Seurat', version = package_version('2.3.0'))
```

**Comment by JohnnyNeuro** [comment]
Hi,
I seem to be having exactly the same issue. 
Looking forward to trying out the new Seurat though!
John

> devtools::install_github(repo = 'satijalab/seurat', ref = 'release/3.0')
Downloading GitHub repo satijalab/seurat@release/3.0
These packages have more recent versions available.
Which would you like to update?
 1:   cli          (1.0.0   -> 1.0.1  ) [CRAN]
 2:   data.table   (1.11.4  -> 1.11.8 ) [CRAN]
 3:   digest       (0.6.15  -> 0.6.18 ) [CRAN]
 4:   dplyr        (0.7.6   -> 0.7.7  ) [CRAN]
 5:   fansi        (0.2.3   -> 0.4.0  ) [CRAN]
 6:   fitdistrplus (1.0-9   -> 1.0-11 ) [CRAN]
 7:   ggplot2      (3.0.0   -> 3.1.0  ) [CRAN]
 8:   ggridges     (0.5.0   -> 0.5.1  ) [CRAN]
 9:   htmlwidgets  (1.2     -> 1.3    ) [CRAN]
10:   later        (0.7.3   -> 0.7.5  ) [CRAN]
11:   mime         (0.5     -> 0.6    ) [CRAN]
12:   openssl      (1.0.1   -> 1.0.2  ) [CRAN]
13:   pkgconfig    (2.0.1   -> 2.0.2  ) [CRAN]
14:   R.utils      (2.6.0   -> 2.7.0  ) [CRAN]
15:   R6           (2.2.2   -> 2.3.0  ) [CRAN]
16:   Rcpp         (0.12.18 -> 0.12.19) [CRAN]
17:   Rdpack       (0.8-0   -> 0.10-1 ) [CRAN]
18:   reticulate   (1.9     -> 1.10   ) [CRAN]
19:   rlang        (0.2.1   -> 0.3.0.1) [CRAN]
20:   scales       (0.5.0   -> 1.0.0  ) [CRAN]
21:   shiny        (1.1.0   -> 1.2.0  ) [CRAN]
22:   tidyr        (0.8.1   -> 0.8.2  ) [CRAN]
23:   tidyselect   (0.2.4   -> 0.2.5  ) [CRAN]
24:   xtable       (1.8-2   -> 1.8-3  ) [CRAN]
25:   zoo          (1.8-3   -> 1.8-4  ) [CRAN]
26:   CRAN packages only
27:   All
28:   None
Enter one or more numbers separated by spaces, or an empty line to cancel
1: 26
cli          (1.0.0   -> 1.0.1  ) [CRAN]
data.table   (1.11.4  -> 1.11.8 ) [CRAN]
digest       (0.6.15  -> 0.6.18 ) [CRAN]
dplyr        (0.7.6   -> 0.7.7  ) [CRAN]
fansi        (0.2.3   -> 0.4.0  ) [CRAN]
fitdistrplus (1.0-9   -> 1.0-11 ) [CRAN]
future       (NA      -> 1.10.0 ) [CRAN]
future.apply (NA      -> 1.0.1  ) [CRAN]
ggplot2      (3.0.0   -> 3.1.0  ) [CRAN]
ggrepel      (NA      -> 0.8.0  ) [CRAN]
ggridges     (0.5.0   -> 0.5.1  ) [CRAN]
globals      (NA      -> 0.12.4 ) [CRAN]
htmlwidgets  (1.2     -> 1.3    ) [CRAN]
later        (0.7.3   -> 0.7.5  ) [CRAN]
listenv      (NA      -> 0.7.0  ) [CRAN]
lsei         (NA      -> 1.2-0  ) [CRAN]
mime         (0.5     -> 0.6    ) [CRAN]
npsurv       (NA      -> 0.4-0  ) [CRAN]
openssl      (1.0.1   -> 1.0.2  ) [CRAN]
pkgconfig    (2.0.1   -> 2.0.2  ) [CRAN]
R.utils      (2.6.0   -> 2.7.0  ) [CRAN]
R6           (2.2.2   -> 2.3.0  ) [CRAN]
Rcpp         (0.12.18 -> 0.12.19) [CRAN]
Rdpack       (0.8-0   -> 0.10-1 ) [CRAN]
reticulate   (1.9     -> 1.10   ) [CRAN]
rlang        (0.2.1   -> 0.3.0.1) [CRAN]
rsvd         (NA      -> 0.9    ) [CRAN]
scales       (0.5.0   -> 1.0.0  ) [CRAN]
shiny        (1.1.0   -> 1.2.0  ) [CRAN]
tidyr        (0.8.1   -> 0.8.2  ) [CRAN]
tidyselect   (0.2.4   -> 0.2.5  ) [CRAN]
xtable       (1.8-2   -> 1.8-3  ) [CRAN]
zoo          (1.8-3   -> 1.8-4  ) [CRAN]
Installing 33 packages: cli, data.table, digest, dplyr, fansi, fitdistrplus, future, future.apply, ggplot2, ggrepel, ggridges, globals, htmlwidgets, later, listenv, lsei, mime, npsurv, openssl, pkgconfig, R.utils, R6, Rcpp, Rdpack, reticulate, rlang, rsvd, scales, shiny, tidyr, tidyselect, xtable, zoo

Error: (converted from warning) unable to access index for repository https://mojaveazure.github.io/loomR/bin/macosx/el-capitan/contrib/3.5:
  cannot open URL 'https://mojaveazure.github.io/loomR/bin/macosx/el-capitan/contrib/3.5/PACKAGES'

> sessionInfo()
R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.14.1

Matrix products: default
BLAS: /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.18      rstudioapi_0.7    magrittr_1.5      usethis_1.4.0     devtools_2.0.1   
 [6] pkgload_1.0.2     R6_2.2.2          rlang_0.2.1       tools_3.5.1       pkgbuild_1.0.2   
[11] sessioninfo_1.1.0 cli_1.0.0         withr_2.1.2       remotes_2.0.2     assertthat_0.2.0 
[16] digest_0.6.15     rprojroot_1.3-2   crayon_1.3.4      processx_3.2.0    callr_3.0.0      
[21] base64enc_0.1-3   fs_1.2.6          ps_1.2.0          curl_3.2          memoise_1.1.0    
[26] glue_1.3.0        compiler_3.5.1    desc_1.2.0        backports_1.1.2   prettyunits_1.0.2
```

**Comment by ChristophH** [comment]
Hi,
it seems like both of you are using Rstudio. Does this error show up if you
try to install Seurat from the terminal (outside of Rstudio)?

**Comment by ltosti** [comment]
Hi there,

I have the same problem when installing from the Terminal.

Thanks for your help!

**Comment by orrzor** [comment]
Hi Christoph,
Thanks for your response. I tried it also in `R` and it did not work. Thank you for helping us with this.
```
> devtools::install_github(repo = 'satijalab/seurat', ref = 'release/3.0')
Downloading GitHub repo satijalab/seurat@release/3.0
Installing 7 packages: future, future.apply, ggrepel, globals, listenv, reticulate, rsvd
Error: (converted from warning) unable to access index for repository https://mojaveazure.github.io/loomR/bin/macosx/el-capitan/contrib/3.5:
  cannot open URL 'https://mojaveazure.github.io/loomR/bin/macosx/el-capitan/contrib/3.5/PACKAGES'
> sessionInfo()
R version 3.5.1 (2018-07-02)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS High Sierra 10.13.6

Matrix products: default
BLAS: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.19      ps_1.2.0          prettyunits_1.0.2 rprojroot_1.3-2  
 [5] digest_0.6.18     crayon_1.3.4      withr_2.1.2       assertthat_0.2.0 
 [9] R6_2.3.0          backports_1.1.2   magrittr_1.5      rlang_0.3.0.1    
[13] cli_1.0.1         curl_3.2          fs_1.2.6          remotes_2.0.2    
[17] callr_3.0.0       desc_1.2.0        devtools_2.0.1    tools_3.5.1      
[21] glue_1.3.0        pkgload_1.0.2     compiler_3.5.1    processx_3.2.0   
[25] base64enc_0.1-3   pkgbuild_1.0.2    sessioninfo_1.1.0 memoise_1.1.0    
[29] usethis_1.4.0    
```

**Comment by andrewwbutler** [comment]
Hi,

We're looking into this but are having trouble consistently reproducing the issue. Do you have your options for warnings set to convert warnings to errors? You can try `options("warn"=0)` to restore the default warning behavior which might allow the install to complete. 

Another alternative to try is to turn off the automatic dependency installation if you already have the required dependencies. 
`install_github(repo = "satijalab/seurat", ref = "release/3.0", dependencies = FALSE)`

**Comment by orrzor** [comment]
Hi Paul and Andrew,

Thanks so much for your replies. The commit to make `loomR` not required allowed me to install Seurat3. Thank you so much, and I hope it is not too much work to track down the loomR issue.

Best, Orr

**Comment by JohnnyNeuro** [comment]
Hi Orr,
How did you work around the loomR requirement? Did you just turn off the automatic dependency installation like Andrew recommended?
John

**Comment by andrewwbutler** [comment]
Hi @JohnnyNeuro,

We just removed the `loomR` requirements on that branch for now while we work out the issue. Try reinstalling with `install_github(repo = "satijalab/seurat", ref = "release/3.0")`

**Comment by JohnnyNeuro** [comment]
Thanks Andrew! That fixed it for me too.
John

**Comment by rspreafico** [comment]
Having the same loomR issue trying to install release 3.0 today

**Comment by mojaveazure** [comment]
Hi @rspreafico,

I am unable to replicate this issue. Can you make sure that you aren't also trying to install suggested packages as well?

```R
> 'loomR' %in% rownames(x = installed.packages())
[1] FALSE
> devtools::install_github(repo = 'satijalab/seurat', ref = 'release/3.0')
Downloading GitHub repo satijalab/seurat@release/3.0
✔  checking for file ‘/tmp/RtmpVknYha/remotes114633220b51/satijalab-seurat-6d361c0/DESCRIPTION’ ...
─  preparing ‘Seurat’:
✔  checking DESCRIPTION meta-information ...
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts
─  checking for empty or unneeded directories
─  looking to see if a ‘data/datalist’ file should be added
─  building ‘Seurat_3.0.0.9000.tar.gz’
   
Installing package into ‘/home/paul/R/x86_64-pc-linux-gnu-library/3.5’
(as ‘lib’ is unspecified)
* installing *source* package ‘Seurat’ ...
** libs
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/Rcpp/include" -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/RcppEigen/include" -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/RcppProgress/include"    -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g -c ModularityOptimizer.cpp -o ModularityOptimizer.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/Rcpp/include" -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/RcppEigen/include" -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/RcppProgress/include"    -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g -c RModularityOptimizer.cpp -o RModularityOptimizer.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/Rcpp/include" -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/RcppEigen/include" -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/RcppProgress/include"    -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g -c RcppExports.cpp -o RcppExports.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/Rcpp/include" -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/RcppEigen/include" -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/RcppProgress/include"    -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g -c data_manipulation.cpp -o data_manipulation.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/Rcpp/include" -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/RcppEigen/include" -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/RcppProgress/include"    -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g -c integration.cpp -o integration.o
g++ -std=gnu++11 -I"/usr/share/R/include" -DNDEBUG  -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/Rcpp/include" -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/RcppEigen/include" -I"/home/paul/R/x86_64-pc-linux-gnu-library/3.5/RcppProgress/include"    -fpic  -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g -c snn.cpp -o snn.o
g++ -std=gnu++11 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o Seurat.so ModularityOptimizer.o RModularityOptimizer.o RcppExports.o data_manipulation.o integration.o snn.o -L/usr/lib/R/lib -lR
installing to /home/paul/R/x86_64-pc-linux-gnu-library/3.5/Seurat/libs
** R
** data
*** moving datasets to lazyload DB
** inst
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded
* DONE (Seurat)
```

**Comment by rspreafico** [comment]
Indeed the way I had to work around it is to explicitly tell `devtools:install_github` not to install suggested packages. This way Seurat installed successfully. For some reason it was pulling suggested packages as well, regardless of whether `loomR` was not installed (my first attemp) or was installed (manually, it installs nicely).

**Comment by mojaveazure** [comment]
Huh, okay. This seems to be a devtools issue, not a Seurat-specific issue.

**Comment by j-andrews7** [comment]
I also ran into this. Might be worth mentioning in the docs.

**Comment by pagarwal14** [comment]
I am getting this error and have not been able to overcome with the suggestions (maybe because I did not understand how to tell devtools explicitly not to install suggested packages (or loomR)?

> withr::with_libpaths(new = "C:/data/installs/R-3.5.3/library", devtools::install_github(repo = 'satijalab/seurat', ref = 'release/3.0'))
Downloading GitHub repo satijalab/seurat@release/3.0
These packages have more recent versions available.
Which would you like to update?

 1:   assertthat (0.2.0 -> 0.2.1) [CRAN]
 2:   cli        (1.0.1 -> 1.1.0) [CRAN]
 3:   colorspace (1.4-0 -> 1.4-1) [CRAN]
 4:   gtable     (0.2.0 -> 0.3.0) [CRAN]
 5:   openssl    (1.2.2 -> 1.3  ) [CRAN]
 6:   rlang      (0.3.1 -> 0.3.3) [CRAN]
 7:   zoo        (1.8-4 -> 1.8-5) [CRAN]
 8:   CRAN packages only
 9:   All
10:   None

Enter one or more numbers separated by spaces, or an empty line to cancel
1: 10
Installing 6 packages: future, future.apply, ggrepel, globals, listenv, rsvd
Error: (converted from warning) unable to access index for repository https://mojaveazure.github.io/loomR/bin/windows/contrib/3.5:
  cannot open URL 'https://mojaveazure.github.io/loomR/bin/windows/contrib/3.5/PACKAGES'
In addition: Warning messages:
1: In untar2(tarfile, files, list, exdir) :
  skipping pax global extended headers
2: In untar2(tarfile, files, list, exdir) :
  skipping pax global extended headers
>

**Comment by AlexChitsazan** [comment]
I too am running into this and non of the above methods are helping

**Comment by mojaveazure** [comment]
@pagarwal14 @AlexChitsazan, you should be able to set `dependencies = NA` or `dependencies = c('Depends', 'Imports')` in the call to `devtools::install_github`

**Comment by j-andrews7** [comment]
@AlexChitsazan @pagarwal14 

You do not need loomR.

@mojaveazure's method didn't work for me (though by all means, it should have). I had to set `dependencies=False` and install the dependencies manually (errors will be thrown telling you which ones are missing):
```
devtools::install_github(repo = 'satijalab/seurat', ref = 'release/3.0', dependencies=FALSE)
```

**Comment by ahdee** [comment]
Hi when I turn off dependencies I'm getting a different error. ```
```
Error in i.p(...) : 
  (converted from warning) installation of package ‘../Seurat_3.0.0.9000.tar.gz’ 
had non-zero exit status
```

```
R version 3.5.3 (2019-03-11)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 7 x64 (build 7601) Service Pack 1
```
Is there anything else I can do? thanks!

**Comment by ahdee** [comment]
nevermind, this was an error because I had to install some of the depenecies manaully.  I did this via biocmanager and it finally worked.

**Comment by kjrgreen1** [comment]
@ahdee 

I am having the same issue and I am very new to R. How did you install the dependencies using biocmanager?

**Comment by ahdee** [comment]
@kjrgreen1 ok assuming you have biocmanager installed.  Do this.  
```
BiocManager::install(c('ape', 'fitdistrplus', 'future', 'future.apply', 'ggridges', 'ica', 'igraph', 'irlba', 'lmtest', 'metap', 'pbapply', 'RANN', 'reticulate', 'ROCR', 'rsvd', 'SDMTools', 'tsne', 'RcppProgress'), ask=T ) 

options("warn"=0)
# some issue iwth loomR
devtools::install_github(repo = 'satijalab/seurat', ref = 'release/3.0', dependencies=FALSE)
```
I got the list of dependencies of the error message.  This works for me.

**Comment by AlexChitsazan** [comment]
This worked for me, thanks!

**Comment by kjrgreen1** [comment]
This worked for me too, thank you!

**Comment by Lesdormis** [comment]
Hi @ahdee, many thanks for your example, But I just tried with your method, and it still has the same error.

Downloading GitHub repo satijalab/seurat@release/3.0
√  checking for file 'C:\Users\19173\AppData\Local\Temp\RtmpmQCOfe\remotes53b867522577\satijalab-seurat-83cfe16/DESCRIPTION'
-  preparing 'Seurat': (406ms)
√  checking DESCRIPTION meta-information ...
-  cleaning src
-  checking for LF line-endings in source and make files and shell scripts (474ms)
-  checking for empty or unneeded directories
-  looking to see if a 'data/datalist' file should be added
-  building 'Seurat_3.0.0.9000.tar.gz'
   
Installing package into �C:/Users/19173/Documents/R/win-library/3.5�
(as �lib� is unspecified)
ERROR: dependencies 'ape', 'igraph', 'irlba', 'lmtest', 'RANN', 'reticulate', 'SDMTools' are not available for package 'Seurat'
* removing 'C:/Users/19173/Documents/R/win-library/3.5/Seurat'
In R CMD INSTALL
installation of package �C:/Users/19173/AppData/Local/Temp/RtmpmQCOfe/file53b81a00cc1/Seurat_3.0.0.9000.tar.gz� had non-zero exit status

And if I manually installed the dependencies again, and it would still occur the error. Do you know what's the issue?

Many thanks

**Comment by ahdee** [comment]
@Lesdormis that is strange.  That error is basically suggesting that those packages are not installed.  So you installed those packagese succseffully?  sometimes you might need restart the Rstudio.  Also try to load each of those seperately and do a ``` sessionInfo()```

**Comment by Lesdormis** [comment]
@ahdee Yep, it works after I restarted the Rstudio. Thanks for your help! Very Appreciated!

**Comment by Emmavb1988** [comment]
Thanks a lot for all the suggestions, installing de packages manually (although after a few rounds of getting a new error and a new uninstalled package) and disabeling the dependencies worked for me as well! Thanks again and thanks for this great package!

**Comment by MichaelStreuber** [comment]
Hi all,

at the end, I was finally able to install Seurat Version 3.0 too and summarized my steps here for other users (thanks a lot to all users from above!!!):

#for Seurat Version3 create a new folder on drive "C:" e.g. "SeuratV3" and provide the path (see below)
#.libPaths("C:/R-libs2")
#.libPaths("C:/R351")
.libPaths("C:/SeuratV3")

#Before you start, "Rtools" has to be installed
#https://cran.r-project.org/bin/windows/Rtools/

#With "sessionInfo()"  you can check which R version and packages you have installed (McSt: "I have now  R version 3.5.3. installed")

##First you have to install "devtools"
install.packages('devtools')
library(devtools)

#Second install "BiocManager" to install specific packages (see below)
install.packages('BiocManager')
library(BiocManager)
BiocManager::install(c('ape', 'fitdistrplus', 'future', 'future.apply', 'ggridges', 'ica', 'igraph', 'irlba', 'lmtest', 'metap', 'pbapply', 'RANN', 'reticulate', 'ROCR', 'rsvd', 'SDMTools', 'tsne', 'RcppProgress'), ask=T )

#You can try options("warn"=0) to restore the default warning behavior which might allow the install to complete
options("warn"=0)

#Now try to install it with depencies = FALSE to turn off the automatic dependency installation if you already have the required dependencies.
devtools::install_github(repo = 'satijalab/seurat', ref = 'release/3.0', dependencies=FALSE)

#You might get this error below
#ERROR: dependencies 'cowplot', 'ggrepel', 'plotly', 'png', 'Rtsne', 'RcppEigen' are not available for package 'Seurat'
#Now install them with BiocManager manually
BiocManager::install(c('cowplot', 'ggrepel', 'plotly', 'png', 'RcppEigen'), ask=T )

#When installing these packages R might ask you if you want to update old packages, yes do it and type "a" for all
BiocManager::install(c('cluster', 'MASS', 'Matrix', 'mgcv', 'survival'), ask=T )
BiocManager::install(c('Rtsne'), ask=T )

#Now it should work when you run the same command from before (some users also recommend to restart R), it could take a while so don't worry
#Install Seurat Version 3.0
devtools::install_github(repo = 'satijalab/seurat', ref = 'release/3.0', dependencies=FALSE)

#Test it
library(Seurat)


Best regards and have fun,
Michael

**Comment by pagarwal14** [comment]
I was able to install with the following commands (in case it helps someone)
The following two deps were needed since the actual installation command gave error for them
install.packages("globals", lib="C:/data/installs/R-3.5.3/library")
install.packages("listenv", lib="C:/data/installs/R-3.5.3/library")
Then the command to install Seurat3
withr::with_libpaths(new = "C:/data/installs/R-3.5.3/library", devtools::install_github(repo = 'satijalab/seurat', ref = 'release/3.0', dependencies = FALSE))

Now can load Seurat3
> library(Seurat, lib.loc = "C:/data/installs/R-3.5.3/library")

To use RunUMAP I had to install UMAP separately (through conda..).
But I get the following error at runtime
> pancreas.integrated <- RunUMAP(object = pancreas.integrated, reduction = "pca", dims = 1:30)
Error in py_call_impl(callable, dots$args, dots$keywords) : 
  Evaluation error: Required version of NumPy not available: installation of Numpy >= 1.6 not found.
> 
The numpy version that I have is 1.16 and that seems to be the latest for Windows.  It was installed when the latest version of conda was installed.

Anyone has experienced this issue?

Thanks,
- Pankaj

**Comment by Dragonmasterx87** [comment]
🤦‍♂️ 

I had all the errors that were in this thread and was like wth is going on here. Then I just did this:

Explanations in #914 on one side and the following run-sequence on one side.
install.packages("Matrix")
install.packages("ggridges")
install.packages("cowplot")
install.packages('devtools')
install.packages('ggrepel')
devtools::install_github(repo = 'satijalab/seurat', ref = 'release/3.0')

Literally that's all you need to do. Is install ggrepel first.
